docs/using/frameworks/flask.rst

```diff
@@ -0,0 +1,196 @@
+defaults to the method of the current request ``flask.request.method``.
+``resource`` must be provided.
+
+.. tip::
+
+    If you aren't familiar with how oso uses actors, actions, and resources to
+    express authorization decisions, see :doc:`/more/glossary` or
+    :doc:`/getting-started/quickstart`.
+
+:py:meth:`flask_oso.FlaskOso.authorize` can be used within route handlers, or in
+the data access layer, depending upon how you want to express authorization.
+Here's a basic example in a route::
+
+    @bp.route("/<int:id>", methods=["GET"])
+    def get_expense(id):
+        expense = Expense.query.get(id)
+        if expense is None:
+            raise NotFound()
+
+        oso.authorize(action="read", resource=expense)
+        return expense.json()
+
+Notice we didn't need to check the return value of ``authorize``.  By default,
+a failed authorization will return a ``403 Forbidden`` response for the current
+request. This can be controlled with
+:py:meth:`flask_oso.FlaskOso.set_unauthorized_action`.
+
+Requiring authorization
+~~~~~~~~~~~~~~~~~~~~~~~
+
+The :py:meth:`flask_oso.FlaskOso.authorize` function provides a flexible API for
+authorization.  ``flask_oso`` does not dictate where this call should occur.
+See :doc:`/getting-started/application/patterns` for more on where oso can be
+integrated.
+
+One downside to requiring routes to call :py:meth:`flask_oso.FlaskOso.authorize`
+explicitly is that it can potentially be forgotten.  To help detect this, the
+:py:meth:`flask_oso.FlaskOso.require_authorization` option can be enabled during
+initialization. This will cause an :py:class:`oso.OsoException` to be raised if
+a call to :py:meth:`flask_oso.FlaskOso.authorize` **is not** made during the
+processing of a request.
+
+Sometimes a route will not need authorization.  To prevent this route from
+causing an authorization error, call
+:py:meth:`flask_oso.FlaskOso.skip_authorization` during request processing:
+
+.. code-block::
+    :emphasize-lines: 18
+
+    oso = Oso()
+    flask_oso = FlaskOso()
+
+    def create_app():
+        app = Flask()
+
+        flask_oso.init_app(app)
+        flask_oso.require_authorization(app)
+
+        oso.load_file("authorization.polar")
+
+        return app
+
+    app = create_app()
+
+    @app.route("/about")
+    def about():
+        flask_oso.skip_authorization()
+        return "about us"
+
+Using decorators
+~~~~~~~~~~~~~~~~
+
+Some developers may prefer a decorator-based API for performing authorization.
+``flask_oso`` provides two decorators:
+
+:py:func:`flask_oso.skip_authorization` marks a route as not requiring
+authorization.  It is the decorator version of
+:py:meth:`flask_oso.FlaskOso.skip_authorization`.
+
+:py:func:`flask_oso.authorize` decorates a route and calls
+:py:meth:`flask_oso.FlaskOso.authorize` before the route body is entered. For
+example::
+
+    from flask_oso import authorize
+
+    @authorize(resource="get_user")
```

Comment:
Can/should this be able to accept any arguments passed into the method? E.g.

```
@authorize(resource=["expense", id])
@app.route("/<id:int>")
def get_expense(id):
    ...
```

Reply:
It's not possible with that particular syntax (the id variable wouldn't be in scope), but we could make it work.

Reply:
Non-blocking and not necessary now.